de.metas.business/src/main/java/de/metas/tax/api/impl/TaxDAO.java

```diff
@@ -196,4 +200,20 @@ public ITranslatableString getTaxCategoryNameById(@Nullable final TaxCategoryId
 		return InterfaceWrapperHelper.getModelTranslationMap(taxCategory)
 				.getColumnTrl(I_C_TaxCategory.COLUMNNAME_Name, taxCategory.getName());
 	}
+
+	@Override
+	public Optional<TaxCategoryId> getTaxCategoryIdByName(@Nullable final String name)
```

Comment:
why accepting null/empty name?

i would suggest to enforce not empty and check it on the caller side